geonode/groups/forms.py

```diff
@@ -86,29 +87,27 @@ class Meta:
 
 
 class GroupMemberForm(forms.Form):
-    user_identifiers = forms.CharField(
+
+    user_identifiers = forms.ModelMultipleChoiceField(
         label=_("User Identifiers"),
-        widget=forms.SelectMultiple(
-            attrs={
-                'class': 'user-select',
-                'style': 'width:300px'
-            }
-        )
+        required=True,
+        queryset=get_user_model().objects.all().exclude(username='AnonymousUser'),
```

Comment:
@afabiani I think we shuld have something like in: https://github.com/GeoNode/geonode/blob/037590a7b06d90614818935758ea390307c7a19d/geonode/base/api/views.py#L128-L139

We want to filter visible users depending on the current user. The logic is: that a user can only see members of public groups (including registered members) and members of the private groups the user is a member of. This is what [get_available_users](https://github.com/GeoNode/geonode/blob/037590a7b06d90614818935758ea390307c7a19d/geonode/people/utils.py#L111) does.

Reply:
@giohappy done